### PR TITLE
feat: [CDS-47674]: adding yaml for tas execution strategy, tas api

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes

1. Added the V2 APIs for the fetching organization and space 
2. Added yaml-snippets for TAS execution strategies for all type of deployment strategies

Test Performed
1. Tested V2 API in PR env and local
2. Tested deployment strategy API returning correct yaml
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/41010)

### CODE REVIEW

File: 125-cd-nextgen/src/main/resources/executionStrategyYaml/tas-bluegreen-with-verify.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          name: "App Setup Step"
+          identifier: setupStep
+          type: TasAppSetup
+          timeout: 10m
+          spec:
+            instanceCount: FromManifest
+            existingVersionToKepp: 3
+            additionalRoutes: ""
+            tempRoutes: ""
+      - step:
```
Comment (RishabhGupta34) [inline 125-cd-nextgen/src/main/resources/executionStrategyYaml/tas-bluegreen-with-verify.yaml:20]: what is this Verify step?

**Review by RishabhGupta34 — COMMENTED** [review]
> Reviewed 12 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41010)* status: 12 file(s) to review. 0 of 2 approvals obtained.

**Review by RishabhGupta34 — DISMISSED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41010)* status: 12 file(s) to review. 1 of 2 approvals obtained.

**Review by piyush-bhuwalka — DISMISSED** [review]
> Reviewed 12 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41010)* status: :shipit: complete! ✓

**Review by piyush-bhuwalka — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41010)* status: 1 of 2 approvals obtained.

**Review by RishabhGupta34 — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41010#-:-NIf08jr1HR-7ls7-pyG:bnfp4nl)* status: :shipit: complete! ✓

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 12 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/41010#-:-NIfDz-a90qUc0RUavyP:bnfp4nl)* status: :shipit: complete! ✓

**Pull request merged**